lib/runtime-core/src/typed_func.rs

```diff
@@ -709,7 +905,10 @@ where
             func_env @ Some(_) => Context::ExternalWithEnv(self.vmctx, func_env),
             None => Context::Internal,
         };
-        let signature = Arc::new(FuncSig::new(Args::types(), Rets::types()));
+        let signature = match self.signature {
+            Some(ref signature) => signature.clone(),
+            None => Arc::new(FuncSig::new(Args::types(), Rets::types())),
+        };
```

Comment:
Is there a reason why we need the `Some` branch here? 🤔 it seems like we already have this information without having to store it at runtime

Reply:
`Args` and `Rets` represent the signature that is infered at compile-time for uniadic host functions. For variadic host functions, the signature is known at runtime, so we need to it in `Func`. I will explain that with a comment.